pkg/kubelet/logs/container_log_manager.go

```diff
@@ -176,19 +176,108 @@ func (c *containerLogManager) Start() {
 	}, logMonitorPeriod)
 }
 
+type containerWithLog struct {
+	ID        string
+	CreatedAt int64
+	logPath   string
+	State     runtimeapi.ContainerState
+}
+
+func (c *containerLogManager) addContainerToMap(container *runtimeapi.Container, path string, containersByName map[string][]containerWithLog) {
+	if container.Metadata != nil {
+		key := container.Metadata.Name + " " + container.PodSandboxId
+		for i := range containersByName[key] {
+			if containersByName[key][i].ID == container.Id {
+				containersByName[key][i].ID = container.Id
+				containersByName[key][i].State = container.State
+				containersByName[key][i].CreatedAt = container.CreatedAt
+				if containersByName[key][i].logPath == "" {
+					containersByName[key][i].logPath = path
+				}
+				return
+			}
+		}
+		withLog := containerWithLog{
+			ID:        container.Id,
+			State:     container.State,
+			CreatedAt: container.CreatedAt,
+			logPath:   path,
+		}
+		containersByName[key] = append(containersByName[key], withLog)
+	}
+}
+
+func (c *containerLogManager) pruneDeadContainerLogs(containersByName map[string][]containerWithLog) {
+	containersToKeep := 2
```

Comment:
kubelet has a containerToKeep config option

Reply:
Here is how the value for containerToKeep is calculated:
```
	klet.containerDeletor = newPodContainerDeletor(klet.containerRuntime, integer.IntMax(containerGCPolicy.MaxPerPodContainer, minDeadContainerInPod))
```
It seems containerGCPolicy.MaxPerPodContainer may carry value greater than 2.
Since this PR doesn't track the size of log files for the dead containers, it would be prudent to use a small value for controlling the space consumption.

I am open to suggestion on the actual value used here.

Reply:
I believe we will need to preserve the containers logs for the calculation of the IntMax that you pasted. Hardcoding it to 2 here doesn't preserve that current behavior.

Reply:
Comment addressed in the next update.